rsocket-core/src/main/java/io/rsocket/RSocketClient.java

```diff
@@ -88,36 +87,25 @@
           started
               .thenMany(Flux.interval(tickPeriod))
               .doOnSubscribe(s -> timeLastTickSentMs = System.currentTimeMillis())
-              .concatMap(i -> sendKeepAlive(ackTimeoutMs, missedAcks))
-              .doOnError(
+              .subscribe(
+                  i -> sendKeepAlive(ackTimeoutMs, missedAcks),
                   t -> {
                     errorConsumer.accept(t);
                     connection.dispose();
-                  })
-              .subscribe();
+                  });
     }
 
-    connection
-        .onClose()
-        .doFinally(
-            signalType -> {
-              cleanup();
-            })
-        .doOnError(errorConsumer)
-        .subscribe();
+    connection.onClose().doFinally(signalType -> cleanup()).subscribe(null, errorConsumer);
 
     connection
         .send(sendProcessor)
```

Comment:
I wrote them this way for readability - operator fusion should be happening so the difference should be negligible.